### PR TITLE
Improve UI elements and update help menu

### PR DESCRIPTION
**Pull request by djleong01**
Improved UI based on PED feedback

- add separator to make UI neater
- add user feedback for command execution
- update help menu 
- update junit tests to reflect adding of "$" to transaction list

### CODE REVIEW

File: src/main/java/longah/commands/HelpCommand.java
Comment (1simjustin) [inline src/main/java/longah/commands/HelpCommand.java:1]: Missing filter command

File: src/main/java/longah/commands/delete/DeleteGroupCommand.java [hunk 1]
```diff
@@ -23,5 +23,6 @@ public DeleteGroupCommand(String commandString, String taskExpression) {
      *
      */
     public void execute(Group group) throws LongAhException {
-        GroupList.deleteGroup(this.taskExpression);}
+        GroupList.deleteGroup(this.taskExpression);
```
Comment (1simjustin) [inline src/main/java/longah/commands/delete/DeleteGroupCommand.java:26]: Can add user feedback to this as well.

File: src/main/java/longah/commands/edit/EditTransactionCommand.java [hunk 2]
```diff
@@ -28,5 +29,6 @@ public void execute(Group group) throws LongAhException {
         transactions.editTransactionList(taskExpression, members);
         group.updateTransactionSolution();
         group.saveAllData();
+        UI.showMessage("Transaction edited successfully!");
```
Comment (1simjustin) [inline src/main/java/longah/commands/edit/EditTransactionCommand.java:32]: Might be good to show what the transaction is.

**Review by 1simjustin — COMMENTED** [review]
In general some of the feedback could do with adding the executed feedback, ie instead of add transaction outputting transaction has been added, can instead do transaction has been added + what the added transaction is

File: src/main/java/longah/commands/delete/DeleteGroupCommand.java (hunk 1 above)
Comment (djleong01) [reply src/main/java/longah/commands/delete/DeleteGroupCommand.java:26]: This is handled within deleteGroup in GroupList itself

File: docs/UserGuide.md [hunk 3]
```diff
@@ -66,8 +66,9 @@ A quick reference table for all commands is presented below. Certain commands ha
     - [Group Balances \& Expense Tracking](#group-balances--expense-tracking)
     - [Debt Simplification](#debt-simplification)
     - [Security](#security)
-    - [Saving the data](#saving-the-data)
-    - [Editing the data file](#editing-the-data-file)
+    - [Data Score](#saving-the-data)
```
Comment (1simjustin) [inline docs/UserGuide.md:69]: What is data score

**Review by 1simjustin — COMMENTED** [review]
User feedback @@

File: src/main/java/longah/commands/HelpCommand.java
Comment (djleong01) [reply src/main/java/longah/commands/HelpCommand.java:1]: added for help menu, will add to UG in a separate PR

**Review by 1simjustin — APPROVED** [review]

**Pull request merged**